fix: Spring Boot 3.5.0 매트릭스에서 querydsl-ktx-spring-boot 테스트 JUnit Platform 버전 불일치

## 문제

main 머지 후 CI의 **Spring Boot 3.5.0 × Java 17** 매트릭스에서 `:querydsl-ktx-spring-boot:test`가 실패한다.

```
Caused by: org.junit.platform.commons.JUnitException:
  OutputDirectoryProvider not available; probably due to unaligned versions of
  the junit-platform-engine and junit-platform-launcher jars on the classpath.
```

다른 매트릭스 (3.0.13, 3.2.12, 3.3.8, 3.4.4)는 통과.

## 원인

Spring Boot 3.5의 `junit-bom`이 지정하는 `junit-platform-engine` 버전과 Gradle 8.12가 기본 번들하는 `junit-platform-launcher` 버전이 맞지 않음.

`querydsl-ktx/build.gradle.kts`에는 `testRuntimeOnly(libs.junit.platform.launcher)`가 있어 BOM이 launcher 버전을 맞춰주지만, **`querydsl-ktx-spring-boot/build.gradle.kts`에는 해당 선언이 없다**. #74에서 실 `@SpringBootTest`를 추가하면서 Spring Boot 3.5 환경에서 문제가 노출됐다.

## 해결

`querydsl-ktx-spring-boot/build.gradle.kts`에 `testRuntimeOnly(libs.junit.platform.launcher)` 추가. `libs.versions.toml`에 이미 정의되어 있다 (버전 없음 = BOM에 위임).

## 작업

- [ ] `querydsl-ktx-spring-boot/build.gradle.kts`에 `testRuntimeOnly(libs.junit.platform.launcher)` 추가
- [ ] 로컬에서 `./gradlew :querydsl-ktx-spring-boot:test -PspringBootVersion=3.5.0` 통과 확인
- [ ] CI 매트릭스 전체 그린 확인

--- a/querydsl-ktx-spring-boot/build.gradle.kts
+++ b/querydsl-ktx-spring-boot/build.gradle.kts
@@ -22,6 +22,11 @@ dependencies {
     }
     testImplementation(libs.h2)
     testImplementation(kotlin("test"))
+
+    // Spring Boot 3.5+의 junit-bom은 Gradle 기본 번들 launcher보다 신 버전을 요구한다.
+    // 명시적으로 testRuntimeOnly로 BOM 관리 버전을 끌어와야 OutputDirectoryProvider
+    // 호환성 이슈를 피할 수 있다.
+    testRuntimeOnly(libs.junit.platform.launcher)
 }
 
 tasks.test {

